libyul/backends/evm/ssa/SSACFG.h

```diff
@@ -183,63 +218,47 @@ class SSACFG
 		std::vector<ValueId> arguments;
 	};
 	struct UnreachableValue {};
-	using ValueInfo = std::variant<UnreachableValue, VariableValue, LiteralValue, PhiValue>;
-	bool isLiteralValue(ValueId const _var) const
-	{
-		return std::holds_alternative<LiteralValue>(valueInfo(_var));
-	}
-	bool isPhiValue(ValueId const _var) const
-	{
-		return std::holds_alternative<PhiValue>(valueInfo(_var));
-	}
-	ValueInfo& valueInfo(ValueId const _var)
-	{
-		return m_valueInfos.at(_var.value);
-	}
-	ValueInfo const& valueInfo(ValueId const _var) const
-	{
-		return valueInfo(_var.value);
-	}
-	ValueInfo const& valueInfo(ValueId::ValueType const _var) const
-	{
-		return m_valueInfos.at(_var);
-	}
 	ValueId newPhi(BlockId const _definingBlock)
 	{
-		ValueId id { m_valueInfos.size() };
-		auto block = m_blocks.at(_definingBlock.value);
-		m_valueInfos.emplace_back(PhiValue{debugDataOf(block), _definingBlock, {}});
-		return id;
+		auto const& block = m_blocks.at(_definingBlock.value);
+		m_phis.emplace_back(PhiValue{debugDataOf(block), _definingBlock, std::vector<ValueId>{}});
+		auto const value = m_phis.size() - 1;
+		yulAssert(value <= std::numeric_limits<ValueId::ValueType>::max());
```

Comment:
```suggestion
		yulAssert(value < std::numeric_limits<ValueId::ValueType>::max());
```
The max value is already a special value denoting "no value", right?